test/include.test.js

```diff
@@ -190,19 +190,322 @@ describe('include', function () {
+                  {id: 1},
+                  {userId: 1},
+                  {title: 'Post A'}
+                ]
+              }
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          user.name.should.equal('User A');
+          user.age.should.equal(21);
+          user.id.should.equal(1);
+          posts = user.posts();
+          posts.length.should.equal(1);
+          posts[0].title.should.equal('Post A');
+          posts[0].userId.should.equal(1);
+          posts[0].id.should.equal(1);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `limit`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              limit: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          user.posts().length.should.equal(1);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `skip`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              skip: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts.length.should.equal(2);
+          posts[0].id.should.equal(2);
+          posts[1].id.should.equal(3);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `offset`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              offset: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts.length.should.equal(2);
+          posts[0].id.should.equal(2);
+          posts[1].id.should.equal(3);
+
+          done();
+        });
+      });
+
+      it('works when using `where` without `limit`, `skip` or `offset`',
+          function(done) {
+        User.findOne({include: {relation: 'posts'}}, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts[0].id.should.equal(1);
+          posts[1].id.should.equal(2);
+          posts[2].id.should.equal(3);
+
+          done();
+        });
+      });
+    });
+
+    context('pagination', function() {
+      it('works when page size is set to the default and `inqlimit` is ' +
+          'exceeded', function(done) {
+        Post.dataSource.settings.inqLimit = 2;
+
+        User.find({include: {relation: 'posts'}}, function(err, users) {
+          if (err) return done(err);
+
+          users.length.should.equal(5);
+
+          delete Post.dataSource.settings.inqLimit;
+
+          done();
+        });
+      });
+
+      it('works when page size is set to 0', function(done) {
+        Post.dataSource.settings.inqLimit = 0;
```

Comment:
Test names says "page size is set to 0", but here you are changing `inqLimit` instead. I find it confusing.